study-builder/studies/pancan/subs.conf

```diff
@@ -167,19 +167,26 @@
     "is_lang_es": """(user.profile.language() == "es")""",
     "is_not_redirect": """
-     !( (user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasOptionStartsWith("C_SARCOMAS_S_OSTEO", "C_BRAIN_")
+     !( (user.profile.language() == "en" &&
+        (user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasOptionStartsWith("C_SARCOMAS_S_OSTEO", "C_BRAIN_")
           || user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_CHILD"].children["PRIMARY_CANCER_CHILD"].answers.hasOptionStartsWith("C_SARCOMAS_S_OSTEO", "C_BRAIN_")
-        )
+        ))
      || ( user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].numChildAnswers("PRIMARY_CANCER_SELF") == 1
             && user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"]
             .answers.hasOptionStartsWith("C_SARCOMAS_ANGIOSARCOMA", "C_GASTRO_ESOPHAGEAL_CANCER", "C_GASTRO_GASTRIC_STOMACH_CANCER", "C_GASTRO_CRC_COLORECTAL", "C_SARCOMAS_S_LEIOMYO")
             && !user.studies["cmi-pancan"].forms["PREQUAL"].questions["DESCRIBE"].answers.hasOption("CHILD_DIAGNOSED")
+            && user.profile.language() == "en"
          )
      || ( user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].numChildAnswers("PRIMARY_CANCER_SELF") == 1
-            && user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasOptionStartsWith("C_BREAST_", "C_GENITOURINARY_S_PROSTATE")
+            && user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasOptionStartsWith("C_BREAST_")
+            && !user.studies["cmi-pancan"].forms["PREQUAL"].questions["DESCRIBE"].answers.hasOption("CHILD_DIAGNOSED")
+            && user.studies["cmi-pancan"].forms["PREQUAL"].questions["ADVANCED_BREAST"].answers.hasOption("YES")
+        )
+     || ( user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].numChildAnswers("PRIMARY_CANCER_SELF") == 1
+            && user.studies["cmi-pancan"].forms["PREQUAL"].questions["PRIMARY_CANCER_LIST_SELF"].children["PRIMARY_CANCER_SELF"].answers.hasOptionStartsWith("C_GENITOURINARY_S_PROSTATE")
             && !user.studies["cmi-pancan"].forms["PREQUAL"].questions["DESCRIBE"].answers.hasOption("CHILD_DIAGNOSED")
-            && (user.studies["cmi-pancan"].forms["PREQUAL"].questions["ADVANCED_BREAST"].answers.hasOption("YES")
-                || user.studies["cmi-pancan"].forms["PREQUAL"].questions["ADVANCED_PROSTATE"].answers.hasOption("YES"))
+            && user.studies["cmi-pancan"].forms["PREQUAL"].questions["ADVANCED_PROSTATE"].answers.hasOption("YES")
+            && user.profile.language() == "en"
```

Comment:
Same question here regarding initializations of language. I HOPE that we can count on it being set to a default, but I don't know. Do you?

Reply:
These users are temp users and while creating temp users, we fall back to english. Verified for users created in DB locally too.
  From CreateTemporaryUserRoute
  
        String languageCode = payload.getLanguageCode();
        if (languageCode == null) {
            //fall back to default language
            languageCode = LanguageStore.DEFAULT_LANG_CODE;
        }